private/bufpkg/bufprotosource/testdata/nested/foo.proto

```diff
@@ -0,0 +1,20 @@
+syntax = "proto3";
+
+message A {
+  message B {
```

Comment:
Hard to break with fewer layers and that is probably because `append` creates a new underlying array more often when the capacity is low.